geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyIndexCommand.java

```diff
@@ -60,32 +62,62 @@ public Result destroyIndex(
 
     if (StringUtils.isBlank(indexName) && StringUtils.isBlank(regionPath)
         && ArrayUtils.isEmpty(group) && ArrayUtils.isEmpty(memberNameOrID)) {
-      return ResultBuilder.createUserErrorResult(
+      return ResultModel.createError(
           CliStrings.format(CliStrings.PROVIDE_ATLEAST_ONE_OPTION, CliStrings.DESTROY_INDEX));
     }
 
     String regionName = null;
     if (regionPath != null) {
       regionName = regionPath.startsWith("/") ? regionPath.substring(1) : regionPath;
     }
-    IndexInfo indexInfo = new IndexInfo(indexName, regionName);
-    indexInfo.setIfExists(ifExists);
+
+    RegionConfig.Index indexInfo = new RegionConfig.Index();
+    indexInfo.setName(indexName);
+    indexInfo.setFromClause(regionName);
+
     Set<DistributedMember> targetMembers = findMembers(group, memberNameOrID);
 
     if (targetMembers.isEmpty()) {
-      return ResultBuilder.createUserErrorResult(CliStrings.NO_MEMBERS_FOUND_MESSAGE);
+      return ResultModel.createError(CliStrings.NO_MEMBERS_FOUND_MESSAGE);
     }
 
     List<CliFunctionResult> funcResults =
         executeAndGetFunctionResult(destroyIndexFunction, indexInfo, targetMembers);
-    Result result = ResultBuilder.buildResult(funcResults);
-    XmlEntity xmlEntity = findXmlEntity(funcResults);
+    ResultModel result = ResultModel.createMemberStatusResult(funcResults, ifExists);
+
+    result.setConfigObject(indexInfo);
 
-    if (xmlEntity != null) {
-      persistClusterConfiguration(result,
-          () -> ((InternalConfigurationPersistenceService) getConfigurationPersistenceService())
-              .deleteXmlEntity(xmlEntity, group));
-    }
     return result;
   }
+
+  @Override
+  public void updateClusterConfig(String group, CacheConfig config, Object element) {
+    RegionConfig.Index indexFromCommand = (RegionConfig.Index) element;
+    String indexName = indexFromCommand.getName();
+
+    String regionName = indexFromCommand.getFromClause();
+    if (regionName != null) {
+      RegionConfig regionConfig = config.findRegionConfiguration(regionName);
+      if (regionConfig == null) {
+        String errorMessage = "Region " + regionName + " not found";
+        if (!ConfigurationPersistenceService.CLUSTER_CONFIG.equals(group)) {
+          errorMessage += " in group " + group;
+        }
+        throw new EntityNotFoundException(errorMessage);
+      }
+
+      if (indexName.isEmpty()) {
+        regionConfig.getIndex().clear();
+      } else {
+        CacheElement.removeElement(regionConfig.getIndex(), indexName);
+      }
+    } else {
+      // Need to search for the index name as region was not specified
+      List<RegionConfig> regions = config.getRegion();
+      for (RegionConfig r : regions) {
+        CacheElement.removeElement(r.getIndex(), indexName);
```

Comment:
CacheElement is Identifiable, so if you override getId() in RegionConfig.Index to return index name then you can use ` CacheElement.removeElement(config.getRegion(), indexName);` which removes an identifiable from a list of identifiables.

Reply:
Maybe I'm missing something - `config.getRegion()` returns a list of `RegionConfig`s but I want to delete a specific index so wouldn't I need to still get the list of indexes with `r.getIndex()` which returns `List<RegionConfig.Index>`?